app/controllers/posts_controller.rb

```diff
@@ -26,6 +27,11 @@ def create
     end
   end
 
+  def destroy
+    @post.destroy
+    redirect_to user_path(User.find(params[:user_id]))
+  end
```

Comment:
👍 Good progress on your project! However, there's an issue that needs attention. Currently, if a post has at least one like or comment, it's not possible to delete the post from both user and admin roles. This could be due to constraints or conditions in your code preventing the deletion.

Here are some steps to address this issue:

- [ ] **Check Constraints:** Review your code to identify any constraints or conditions that might be preventing the deletion of a post with likes or comments. Look for conditions in your controller or service that might be restricting the deletion based on associations.

- [ ] **Handle Deletions Appropriately:** Ensure that your deletion logic is handling associated records (likes, comments, etc.) appropriately before attempting to delete the post. You may need to destroy or nullify associated records first.

![image](https://github.com/mahdinoori2000/blog-posts/assets/55840999/3114bb57-471d-47bf-9607-61ca22ee6813)

![image](https://github.com/mahdinoori2000/blog-posts/assets/55840999/73da399d-9cb2-4376-9ee1-3a5ae04b25df)

By addressing this issue, you'll enhance the overall usability and robustness of your application. Keep up the good work, and feel free to reach out if you have any questions or need further assistance! 🚀🛠️

Reply:
- [ ] This issue has not been addressed. When I try to delete a `Post` instance that has either likes or comments, the error still shows:

![image](https://github.com/mahdinoori2000/blog-posts/assets/77106962/290a87ff-84fb-4933-9e4b-af387da66012)

---

To fix this, I would advise you to delete the posts and their dependents. You can define this in the `app/models/post.rb` file. Something like this:

```ruby
has_many :likes, dependent: :destroy
has_many :comments, dependent: :destroy
```